### ISSUE TITLE
NumberInput does not consider LocaleProvider's locale

### ISSUE THREAD

**Issue by heikkilae** [chakra-ui/ark#3133]
### Description

The NumberInput component does not respect the locale provided by the LocaleProvider. The issue arises because the use-number-input hook only uses the dir from the useLocaleContext() but fails to include the locale property.

### Fix Proposal

This patch fixes the issue where the `locale` property is not considered in the `NumberInput` component.

```diff --git a/packages/react/src/components/number-input/use-number-input.ts b/packages/react/src/components/number-input/use-number-input>index bd9f99a6a..29288ff11 100644
--- a/packages/react/src/components/number-input/use-number-input.ts
+++ b/packages/react/src/components/number-input/use-number-input.ts
@@ -18,7 +18,7 @@ export interface UseNumberInputReturn extends numberInput.Api<PropTypes> {}
 
 export const useNumberInput = (props: UseNumberInputProps = {}): UseNumberInputReturn => {
   const { getRootNode } = useEnvironmentContext()
-  const { dir } = useLocaleContext()
+  const { locale, dir } = useLocaleContext()
   const field = useFieldContext()
 
   const initialContext: numberInput.Context = {
@@ -31,6 +31,7 @@ export const useNumberInput = (props: UseNumberInputProps = {}): UseNumberInputR
     readOnly: field?.readOnly,
     required: field?.required,
     invalid: field?.invalid,
+    locale,
     dir,
     getRootNode,
     value: props.defaultValue,
```

### Link to Reproduction (or Detailed Explanation)

https://codesandbox.io/p/devbox/brave-sky-2rc3hw

### Steps to Reproduce

**Steps:**
1. Enter the value `1450.50` into the NumberInput input field.
2. Press enter
3. Observe that the NumberInput does not reflect the provided locale settings for number formatting or direction.

**Expected Behavior:**
If the locale is set to fi-FI, the expected input formatting for a value like `1450.50` should be displayed as `1 450,50 €` with the correct decimal and thousands separators, in line with Finnish formatting conventions.

**Actual Behavior:**
Instead of displaying the value as 1 450,50 (with a space as the thousands separator and a comma as the decimal separator, as expected in Finnish formatting), it keeps the value as `€1,450.50` (with a comma as the thousands separator and a dot as the decimal separator, which is common in English locales like en-EN).

### Ark UI Version

4.4.4

### Framework

- [X] React
- [ ] Solid
- [ ] Vue

### Browser

Chrome 131.0.6778.109 (Official Build) (64-bit)

**Comment by cschroeter** [comment]
@heikkilae  Thanks mate for taking your time to create this issue. Fix is available in 568fb8ee6e030f0372b10b1096c4606c6acb3095

**Comment by heikkilae** [comment]
@cschroeter: Wow, that was fast! Can you give an ETA for the new release? I’m looking forward to seeing the fix in Chakra.
